config/routes.rb

```diff
@@ -11,7 +11,8 @@
     match 'create', to: 'bigbluebutton_api#create', via: [:get, :post]
     get 'end', to: 'bigbluebutton_api#end'
     get 'join', to: 'bigbluebutton_api#join'
-    get 'getRecordings', to: 'bigbluebutton_api#get_recordings', as: :get_recordings
+    get '*Recordings', to: 'bigbluebutton_api#recordings_disabled', constraints: RecordingConstraint, via: :all
```

Comment:
Rather than use a constraint here, which has to be checked every request, why don't you put a conditional in the routes definition, so it defines different routes depending on whether the DB is present?

```ruby
if 'true'.casecmp?(ENV['DB_DISABLED'])
  get 'getRecordings', to: 'bigbluebutton_api#get_recordings_disabled', as: :get_recordings
  …
else
  get 'getRecordings', to: 'bigbluebutton_api#get_recordings', as: :get_recordings
  …
end
```

I think you might have to use separate controllers for each API even in the disabled case, since the return format expected is different on the different APIs.

The `getRecordings` API has to return an `<returncode>SUCCESS</returncode>`, and an empty `<recordings>` element and `noRecordings` messageKey in the case where there are no recordings returned

The other APIs must return a `<returncode>FAILED</returncode>` and a messageKey of `notFound`.